### PR TITLE
Add CI, Autoformat, and Delegate Floating Point Parsing

### PR DESCRIPTION
**Pull request by stephencelis**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,9 +1,12 @@
-PLATFORM_IOS = iOS Simulator,name=iPhone 11 Pro Max
+PLATFORM_IOS = iOS Simulator,name=iPhone 11 Pro
 PLATFORM_MACOS = macOS
 PLATFORM_TVOS = tvOS Simulator,name=Apple TV 4K (at 1080p)
 
 default: test
 
+benchmarks:
+	swift run swift-parsing-benchmark --allow-debug-build
```
Comment (mbrandonw) [inline Makefile:8]:
```suggestion
	swift run -c release swift-parsing-benchmark
```

File: Sources/swift-parsing-benchmark/main.swift [hunk 2]
```diff
@@ -9,15 +9,15 @@ Benchmark.main(
 //    boolSuite,
 //    colorSuite,
 //    csvSuite,
-    dateSuite,
+//    dateSuite,
 //    httpSuite,
 //    jsonSuite,
-//    numericsSuite,
```
Comment (mbrandonw) [inline Sources/swift-parsing-benchmark/main.swift:15]: we should bring back on the benchmarks now

File: Tests/ParsingTests/DoubleTests.swift [hunk 3]
```diff
@@ -50,8 +50,8 @@ final class DoubleTests: XCTestCase {
     XCTAssertEqual(" Hello", String(input))
 
     input = "1234567890123456789012345678901234567890 Hello"[...].utf8
-    XCTAssertEqual(.infinity, parser.parse(&input))
-    XCTAssertEqual(" Hello", String(input))
+    XCTAssertEqual(nil, parser.parse(&input))
+    XCTAssertEqual("1234567890123456789012345678901234567890 Hello", String(input))
```
Comment (mbrandonw) [inline Tests/ParsingTests/DoubleTests.swift:54]: this is an expected change right?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
+name: CI
+
+on:
+  push
+  # :
+  #   branches:
+  #     - main
+  # pull_request:
+  #   branches:
+  #     - '*'
```
Comment (mbrandonw) [inline .github/workflows/ci.yml]: we should uncomment these too

File: Tests/ParsingTests/DoubleTests.swift (hunk 3 above)
Comment (stephencelis) [reply Tests/ParsingTests/DoubleTests.swift:54]: Yeah. Overflowing to infinity is probably not what we want anyway.

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    branches:
+      - '*'
+
+jobs:
+  tests:
+    runs-on: macOS-latest
+    strategy:
+      matrix:
+        xcode:
+          - 12.3
```
Comment (mbrandonw) [inline .github/workflows/ci.yml:17]: do we want some Xcode 11s too?

File: README.md [hunk 6]
```diff
@@ -334,9 +333,9 @@ Numerics.Comma separated: Int.parser           5390299.000 ns ±   5.52 %
 Numerics.Comma separated: Scanner.scanInt     86885256.500 ns ±   5.40 %         16
 Numerics.Comma separated: String.split       120884645.000 ns ±   2.18 %         11
 Numerics.Double.init                                62.000 ns ± 615.58 %    1000000
-Numerics.Double.parser                              60.000 ns ± 640.87 %    1000000
+Numerics.Double.parser                              92.000 ns ± 249.73 %    1000000
```
Comment (stephencelis) [inline README.md:336]: We were using `strtod_l` and family before, which is what Swift uses under the hood and is just as fast as `Double.init`, but CI was failing to parse due to not being able to get a handle on contiguous memory. May be able to recover this again in the future, but let's be safe for now.

**Review by mbrandonw — APPROVED** [review]

**Pull request merged**